Math/NumberTheory/Zeta/Dirichlet.hs

```diff
@@ -0,0 +1,212 @@
+    pisAndFacs' :: [[a]]
+    pisAndFacs' = scanl (flip (:)) [] pisAndFacs
+
+    -- Second summand of RHS in (12) for k = [1 ..]
+    rhs2 :: [a]
+    rhs2 = zipWith (*) (cycle [-1, 1]) $ map (sum . zipWith (*) zets) pisAndFacs'
+
+    -- [pi^3 / (2^4), pi^5 / (2^6), pi^7 / (2^8) ..]
+    -- Second factor of third addend in RHS of (12).
+    pis :: Floating a => [a]
+    pis = map approximateValue $ zipWith Exact (tail odds) (map recip (iterate (4 *) 16))
+
+    -- [[3!, 5!, 7! ..], [5!, 7! ..] ..]
+    oddFacs :: [[a]]
+    oddFacs = iterate tail (tail factorial1)
+
+    -- [1, 4, 16 ..]
+    fours :: [a]
+    fours = iterate (4 *) 1
+
+    -- [[3! * 2^0, 5! * 2^2, 7! * 2^4 ..], [5! * 2^0, 7! * 2^2, 9! * 2^4 ..] ..]
+    infSumDenoms :: [[a]]
+    infSumDenoms = map (zipWith (*) fours) oddFacs
+
+    -- [pi^0, pi^2, pi^4, pi^6 ..]
+    pis2 :: [a]
+    pis2 = map approximateValue $ zipWith Exact evens (repeat 1)
+
+    -- [pi^0 * E_1(1), - pi^2 * E_3(1), pi^4 * E_5(1) ..]
+    infSumNum :: [a]
+    infSumNum = zipWith3 (\sgn p eulerP -> sgn * p * eulerP) (cycle [1, -1])
+                                                             pis2
+                                                             (map fromRational . skipEvens $ eulerPolyAt1)
+
+    -- [     [ pi^0 * E_1(1)  (-1) * pi^2 * E_3(1)   ]      [ (-1) * pi^2 * E_3(1)  pi^4 * E_5(1)    ]      [ pi^4 * E_5(1)  (-1) * pi^6 * E_7(1)    ]  ]
+    -- | sum | -------------, -------------------- ..|, sum | --------------------, ------------- .. |, sum | -------------, -------------------- .. |..|
+    -- [     [       3!                 5!           ]      [          5!                 7!         ]      [       7!                9!             ]  ]
+    infSum :: [a]
+    infSum = map (suminf eps . zipWith (/) infSumNum) infSumDenoms
+
+    -- Third summand of the right hand side of (12).
+    rhs3 :: [a]
+    rhs3 = zipWith3 (\sgn p inf -> sgn * p * inf) (cycle [-1, 1])
+                                                  pis
+                                                  infSum
+
+-- | Infinite sequence of approximate (up to given precision)
+-- values of Dirichlet beta-function at integer arguments, starting with @β(0)@.
+-- The algorithm used to compute @β@ for even arguments was derived from
+-- <https://arxiv.org/pdf/0910.5004.pdf An Euler-type formula for β(2n) and closed-form expressions for a class of zeta series>
+-- by F. M. S. Lima, formula (12).
+--
+-- > > take 5 (betas 1e-14) :: [Double]
+-- > [0.5,0.7853981633974483,0.9159655941772191,0.9689461462593693,0.988944551741105]
+betas :: (Floating a, Ord a) => a -> [a]
+betas eps = e : o : scanl1 f (intertwine es os)
+  where
+    e : es = betasEven eps
+    o : os = betasOdd'
+
+    intertwine (x : xs) (y : ys) = x : y : intertwine xs ys
+    intertwine xs ys = xs ++ ys
+
+    -- Cap-and-floor to improve numerical stability:
+    -- 0 < zeta(n + 1) - 1 < (zeta(n) - 1) / 2
```

Comment:
Should be beta instead of zeta. 
Also, the second part of inequality becomes not that trivial, as it was for zeta. I'll check if it is still true.

Reply:
I mistakenly left the comment as it was, thought my notes on paper say something else. I think I got `1 > beta(n + 1) - 1 > (beta(n) - 1) / 2`